src/software/ai/hl/stp/tactic/patrol_tactic_test.cpp

```diff
@@ -0,0 +1,289 @@
+#include "software/ai/hl/stp/tactic/patrol_tactic.h"
+
+#include <gtest/gtest.h>
+
+#include "software/ai/hl/stp/action/move_action.h"
+#include "software/test_util/test_util.h"
+
+class PatrolTacticTest : public testing::Test
+{
+   protected:
+    std::vector<RobotState> robotStateToCompleteActions;
+    std::vector<std::shared_ptr<MoveAction>> expectedActions;
+    const double COST_OF_ASSIGNED_ROBOT                      = 0.0;
+    const double COST_OF_NONASSIGNED_ROBOT_UNASSIGNED_TACTIC = 1.0;
+
+    /*
+     * populates the expectedActions test variable in the order of the patrol points
+     */
+    void GenerateExpectedActions(Robot robot, std::vector<Point> patrolPoints,
+                                 Angle angle, double speedAtPatrolPoints)
+    {
+        for (int i = 0; i < patrolPoints.size(); i++)
+        {
+            auto expected_action = std::make_shared<MoveAction>(false);
+            expected_action->updateControlParams(
+                robot, patrolPoints[i], angle, speedAtPatrolPoints, DribblerEnable::OFF,
+                MoveType::NORMAL, AutokickType::NONE, BallCollisionType::AVOID);
+            expectedActions.push_back(expected_action);
+        }
+    }
+
+    /*
+     *  populates the robotStateToCompleteActions test variable in the order of patrol
```

Comment:
This comment style seems different to ones below such as "simulateActionToCompletion". It'd be best if they all adhered to a consistent style similar to simulateActionToCompletion where there is a section on what the function does and @params included.